trade-finance-manager-ui/server/controllers/utilisation-reports/index.ts

```diff
@@ -28,3 +29,35 @@ export const getUtilisationReports = async (req: Request, res: Response) => {
     return res.render('_partials/problem-with-service.njk', { user });
   }
 };
+
+export const getUtilisationReportByBankId = async (req: Request, res: Response) => {
+  const { userToken, user } = req.session;
+  const { bankId } = req.params;
+
+  const errors = validationResult(req);
+  if (!errors.isEmpty()) {
+    console.error(`Error rendering utilisation for bank with id ${bankId}:`, errors.array());
+    return res.render('_partials/problem-with-service.njk', { user });
+  }
+
+  try {
+    const submissionMonth = getIsoMonth(new Date());
+    const reconciliationSummaryItems = await api.getUtilisationReportsReconciliationSummary(submissionMonth, asString(userToken));
+    const bank = reconciliationSummaryItems.find((summaryItem) => summaryItem.bank.id === bankId)?.bank;
+    if (!bank) {
+      throw new Error(`Bank with id ${bankId} not found`);
+    }
+
+    const reportPeriod = getFormattedReportPeriod();
+
+    return res.render('utilisation-reports/utilisation-report-for-bank.njk', {
+      user,
+      activePrimaryNavigation: 'utilisation reports',
```

Comment:
It might be nice to have these values pulled from a constant (we could just include `'utilisation reports'` initially).

Reply:
I have introduced this as part of a separate branch as I may as well introduce this constant across the project - will link the new PR in this comment thread once it is up

Reply:
It's here: #2552